src/Nethermind/Nethermind.Synchronization/ParallelSync/MultiSyncModeSelector.cs

```diff
@@ -86,6 +86,11 @@ public class MultiSyncModeSelector : ISyncModeSelector, IDisposable
         public event EventHandler<SyncModeChangedEventArgs>? Changing;
         public event EventHandler<SyncModeChangedEventArgs>? Changed;
 
+        // Temporary event and variable used for removing snap capability after SnapSync finish.
+        // Will be removed after implementing missing functionality - serving data via snap protocol.
+        public event EventHandler<EventArgs> SnapSyncFinished;
```

Comment:
Same here

Reply:
If we use `Changed`, then in `SnapCapabilitySwitcher` we will need to check `SyncMode current` for `Full` every time when this event is invoked and disable when `Full` will be there first time. It can be invoked a lot of times during sync and a lot of unnecessary calculations will be needed. Using new event `SnapSyncFinished` it is invoked only once - exactly when is needed. And it will be removed when Snap serving will be implemented

Reply:
I'd recommend still using `Changed`. The performance implication is likely insignificant (it does not get called a lot in the grand scheme of things, and consistent branch is easy to predict by CPU), and its cleaner IMO.

Reply:
DIsabling is easy - just unsubscribe from the event.

Reply:
Ok, I will change it